cmd/snap-update-ns/main.go

```diff
@@ -77,7 +77,7 @@ func run() error {
 		return fmt.Errorf("cannot open lock file for mount namespace of snap %q: %s", snapName, err)
 	}
 	defer lock.Close()
-	if err := lock.Lock(); err != nil {
+	if err := lock.Lock(0); err != nil {
```

Comment:
Well, I think I've often seen this kind of thing called `TryLock`, with a signature like `TryLock() error`, returning some fixed error when unable to lock. We even had an implementation of this in our previous incarnation of file locks (look at 2d6bdcd5a5c794463a1d5ec4115c2bd0d5550b42 )